### DISCUSSION TITLE
加载 hdr 文件 场景生效 无环境光

### DISCUSSION THREAD

**Discussion by MrTangmx** [galacean/discussion#109] [Engine] [answered]
`import { OrbitControl } from "@galacean/engine-toolkit-controls";
import { AssetType, BackgroundMode, Camera, GLTFResource, PrimitiveMesh, SkyBoxMaterial, TextureCube, WebGLEngine } from "@galacean/engine";

WebGLEngine.create({ canvas: "canvas" }).then((engine) => {
  engine.canvas.resizeByClientSize();

  const rootEntity = engine.sceneManager.activeScene.createRootEntity();

  const cameraEntity = rootEntity.createChild("camera");
  cameraEntity.addComponent(Camera);
  cameraEntity.transform.setPosition(3, 3, 3);
  cameraEntity.addComponent(OrbitControl);

 const scene = engine.sceneManager.activeScene
 scene.ambientLight.diffuseSolidColor.set(
    1,
    1,
    1,
    1
  );
  engine.resourceManager.load({ type: AssetType.HDR, url: '/kloofendal_43d_clear_1k.hdr' }).then( texture => {
	const sky = scene.background.sky;
	const skyMaterial = new SkyBoxMaterial(engine);
	scene.background.mode = BackgroundMode.Sky;
	sky.material = skyMaterial;
	sky.mesh = PrimitiveMesh.createCuboid(engine, 1, 1, 1);
	skyMaterial.texture = texture;
	skyMaterial.textureDecodeRGBM = true;
	engine.run();
});
  engine.resourceManager
    .load<GLTFResource>(
      "/03.1695866210938.glb"
    )
    .then((gltf) => {
      rootEntity.addChild(gltf.defaultSceneRoot);
    });

  engine.run();
});
`

`"@galacean/engine": "^1.0.4",
    "@galacean/engine-toolkit": "^1.0.0",
    "@galacean/engine-toolkit-controls": "^1.0.0",
    "@galacean/engine-toolkit-framebuffer-picker": "^1.0.0"`

**Comment by eyworldwide** [comment]
`.hdr ` 文件经过 gltfViewer 处理了吗？参考文档：https://galacean.antgroup.com/#/docs/latest/cn/ambient-light

**Reply by zhuxudong** [reply]
不用烘焙，他只是想当作背景，从代码上来看跟案例的是一样的，除了 URL 是相对路径, @MrTangmx  你试试官网的 url 呢：https://gw.alipayobjects.com/os/bmw-prod/b578946a-8a25-4543-8161-fa92f92ae1ac.bin

**Reply by MrTangmx** [reply]
我需要的是这个hdr中的环境光，背景可以不用

**Reply by MrTangmx** [reply]
正在尝试烘培之后的env文件是否可用

**Reply by zhuxudong** [reply, marked as answer]
那你 loader 不应该使用 HDRLoader 呢，应该使用 `AssetType.Env`,参考[官网案例](https://galacean.antgroup.com/#/examples/latest/ambient-light)
```ts
 engine.resourceManager
    .load<AmbientLight>({
      type: AssetType.Env,
      url: "https://gw.alipayobjects.com/os/bmw-prod/6470ea5e-094b-4a77-a05f-4945bf81e318.bin",
    })
    .then((ambientLight) => {
      scene.ambientLight = ambientLight;
    });
```

然后这个烘焙产物是需要经过编辑器或者 glTF-viewer 离线烘焙得到的，参考[文档](https://galacean.antgroup.com/#/docs/latest/cn/ambient-light)

**Reply by zhuxudong** [reply]
你先自己试试，还不行的话你可以把 HDR 发我，我烘焙好发你 url

**Reply by MrTangmx** [reply]
感谢帮助，我先试试

**Reply by MrTangmx** [reply]
可以应用到环境光，感谢大佬们帮助
